internal/terraform/eval_conditions.go

```diff
@@ -49,121 +27,158 @@ func (c checkType) FailureSummary() string {
 //
 // If any of the rules do not pass, the returned diagnostics will contain
 // errors. Otherwise, it will either be empty or contain only warnings.
-func evalCheckRules(typ checkType, rules []*configs.CheckRule, ctx EvalContext, self addrs.Referenceable, keyData instances.RepetitionData, diagSeverity tfdiags.Severity) (diags tfdiags.Diagnostics) {
+func evalCheckRules(typ addrs.CheckType, rules []*configs.CheckRule, ctx EvalContext, self addrs.Checkable, keyData instances.RepetitionData, diagSeverity tfdiags.Severity) tfdiags.Diagnostics {
```

Comment:
An interesting quirk here is that in most other cases we have the `configs`-package objects be aware of the module-local local part of their address, so that the `terraform` package can just absolute-ize it by adding the module instance address.

In this case I guess check blocks don't really have addresses until we evaluate them, which is a bit different but I suppose not a big deal.

Overall this inconsistency seems reasonable since it avoids the complexity of defining both local and absolute variants of the address type, where we wouldn't otherwise have any reason to use a non-absolute check rule address.